### PR TITLE
uwsgi: 2.0.20 -> 2.0.21

### PR DESCRIPTION
**Pull request by gdamjan**
removed `missing-arginfo-php8.patch`, since it is already included in the 2.0.21 release
https://uwsgi-docs.readthedocs.io/en/latest/Changelog-2.0.21.html

additionally, when php is enabled, make sure uwsgi loads the
...php-with-extensions-…/lib/php.ini file
so that it will load all the extensions.

lastly, change the source url to the github tag archive,
it seems the project web site is not that well maintained.

tested a simple `phpinfo()` page with both 8.1.11 and 8.0.24.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gdamjan** [comment]
I've reached to the uwsgi project to see if the removal of the tarball from their web page is not intentional, or we should start using the github provided archive of the release tag.

so let's keep it a draft pr for a day or two.

**Review by winterqt — CHANGES_REQUESTED** [review]
Please fix the commit message per the [manual](https://nixos.org/manual/nixpkgs/stable/#submitting-changes-making-patches) -- specifically, mention the previous version.

File: pkgs/servers/uwsgi/default.nix [hunk 1]
```diff
@@ -59,20 +59,19 @@ in
 
 stdenv.mkDerivation rec {
   pname = "uwsgi";
-  version = "2.0.20";
+  version = "2.0.21";
 
   src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/servers/uwsgi/default.nix:64]: This should be fetchFromGitHub

Comment (gdamjan) [reply pkgs/servers/uwsgi/default.nix]: done.

File: pkgs/servers/uwsgi/default.nix [hunk 2]
```diff
@@ -59,20 +60,21 @@ in
 
 stdenv.mkDerivation rec {
   pname = "uwsgi";
-  version = "2.0.20";
+  version = "2.0.21";
 
-  src = fetchurl {
```
Comment (SuperSandro2000) [inline pkgs/servers/uwsgi/default.nix:64]: You forgot to remove fetchurl from line 1

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM but can't change that from github web

File: pkgs/servers/uwsgi/default.nix (hunk 2 above)
Comment (gdamjan) [reply pkgs/servers/uwsgi/default.nix:64]: indeed. fixed

**Pull request merged**